src/stream-tcp-list.c

```diff
@@ -890,6 +895,53 @@ void StreamTcpPruneSession(Flow *f, uint8_t flags)
     SCReturn;
 }
 
+/**
+ * \brief Adds the following information to the TcpSegment from the current
+ *  packet being processed: time values, packet length, pcap_cnt, and the
+ *  header data of the packet. This information is added to the TcpSegment so
+ *  that it can be used in pcap capturing (log-pcap-stream) to dump the tcp
+ *  session at the beginning of the pcap capture.
+ * \param seg TcpSegment where information is being stored.
+ * \param p Packet being processed.
+ * \param tv Thread-specific variables.
+ * \param ra_ctx TcpReassembly thread-specific variables
+ */
+static void StreamTcpSegmentAddPacketData(TcpSegment *seg, Packet *p,
+        ThreadVars *tv, TcpReassemblyThreadCtx *ra_ctx)
+{
+    if (seg->pcap_hdr_storage == NULL || seg->pcap_hdr_storage->pkt_hdr == NULL) {
+        return;
+    }
+    if (GET_PKT_DATA(p) != NULL && GET_PKT_LEN(p) > p->payload_len) {
+        seg->pcap_hdr_storage->ts.tv_sec = p->ts.tv_sec;
+        seg->pcap_hdr_storage->ts.tv_usec = p->ts.tv_usec;
+        seg->pcap_hdr_storage->pktlen = GET_PKT_LEN(p);
+        seg->pcap_hdr_storage->pcap_cnt = p->pcap_cnt;
```

Comment:
`Packet::pcap_cnt` is only going to be set when we're reading a pcap. For live traffic modes it isn't used.

Reply:
This is a good point that I hadn't realized as my development was tested against offline pcaps the whole time. The pcap_cnt member is only being used to determine the order of dumping, i.e. seeing which pcap was older. Working out a way to do this that will work for both live and offline modes shouldn't be an issue.